pkg/template/processor.go

```diff
@@ -0,0 +1,193 @@
+	}
+	return createdOrUpdated, nil
+}
+
+func (p Processor) createOrUpdateObj(newResource runtime.Object, forceUpdate bool, owner v1.Object) (bool, error) {
+	// gets the meta accessor to the new resource
+	metaNew, err := meta.Accessor(newResource)
+	if err != nil {
+		return false, errors.Wrapf(err, "cannot get metadata from %+v", newResource)
+	}
+
+	// creates a deepcopy of the new resource to be used to check if it already exists
+	existing := newResource.DeepCopyObject()
+
+	// set current object as annotation
+	//newWithoutAnnotation := newResource.DeepCopyObject()
+	annotations := metaNew.GetAnnotations()
+	newConfiguration := getNewConfiguration(newResource)
+	if annotations == nil {
+		annotations = map[string]string{}
+	}
+	annotations[LastAppliedConfigurationAnnotationKey] = newConfiguration
+	metaNew.SetAnnotations(annotations)
+
+	// gets current object (if exists)
+	namespacedName := types.NamespacedName{Namespace: metaNew.GetNamespace(), Name: metaNew.GetName()}
+	if err := p.cl.Get(context.TODO(), namespacedName, existing); err != nil {
+		creationErr := p.createObj(newResource, metaNew, owner)
+		if apierrors.IsNotFound(err) || creationErr == nil {
+			return true, creationErr
+		}
+		return false, errors.Wrapf(creationErr, "unable to get the resource '%v' and creation failed too: %s", existing, creationErr.Error())
+	}
+
+	// gets the meta accessor to the existing resource
+	metaExisting, err := meta.Accessor(existing)
+	if err != nil {
+		return false, errors.Wrapf(err, "cannot get metadata from %+v", existing)
+	}
+
+	// as it already exists, check using the UpdateStrategy if it should be updated
+	if !forceUpdate {
+		existingAnnotations := metaExisting.GetAnnotations()
+		if existingAnnotations != nil {
+			if newConfiguration == existingAnnotations[LastAppliedConfigurationAnnotationKey] {
+				return false, nil
+			}
+		}
+	}
+
+	// retrieve the current 'resourceVersion' to set it in the resource passed to the `client.Update()`
+	// otherwise we would get an error with the following message:
+	// "nstemplatetiers.toolchain.dev.openshift.com \"basic\" is invalid: metadata.resourceVersion: Invalid value: 0x0: must be specified for an update"
+	originalGeneration := metaExisting.GetGeneration()
+	metaNew.SetResourceVersion(metaExisting.GetResourceVersion())
+	if err := p.cl.Update(context.TODO(), newResource); err != nil {
+		return false, errors.Wrapf(err, "unable to update the resource '%v'", newResource)
+	}
+
+	// gets the meta accessor to the resource that was updated
+	metaNewAfterUpdate, err := meta.Accessor(newResource)
+	if err != nil {
+		return false, errors.Wrapf(err, "cannot get metadata from %+v", newResource)
+	}
+
+	// check if it was changed or not
+	return originalGeneration == metaNewAfterUpdate.GetGeneration(), nil
+}
+
+func getNewConfiguration(newResource runtime.Object) string {
+	newYaml := marshalObjectContent(newResource)
```

Comment:
I'm not sure the `marshalObjectContent` alone makes sense. Shouldn't we move its code here in `getNewConfiguration`?

Reply:
I believe that after these changes 8720488 it makes  sense to keep it in separated function